libraries/AP_Notify/Display.cpp

```diff
@@ -404,15 +404,46 @@ void Display::update()
 
 void Display::update_all()
 {
-    update_text(0);
-    update_mode(1);
-    update_battery(2);
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 0)) {
+        update_text(0);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 1)) {
+        update_mode(1);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 2)) {
+        update_battery(2);
+    }
+
 #if AP_GPS_ENABLED
-    update_gps(3);
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 3)) {
+        update_gps(3);
+    }
 #endif
-    //update_gps_sats(4);
-    update_prearm(4);
-    update_ekf(5);
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 4)) {
+        //update_gps_sats(4);
+        update_prearm(4);
+    }
+
+    if(!BIT_IS_SET(pNotify->_send_text_src_override, 5)) {
+        update_ekf(5);
+    }
+}
+
+void Display::send_text(const char *text, uint8_t r)
+{
+    if (nullptr == text) {
+        return;
+    }
+    if (r >= NOTIFY_TEXT_NUM_ROWS) {
+        return;
+    }
+    char txt [NOTIFY_TEXT_BUFFER_SIZE] = {};
+    memset(txt, ' ', NOTIFY_TEXT_BUFFER_SIZE);
```

Comment:
I don't think you need this? Or maybe the display does not like null characters?

Reply:
Yeah they are there to clear the display. Say if it was displaying "hello" and i send "hi" without that the display will display "hillo".